File: src/main/java/com/autosale/shop/service/AuthenticationService.java
package com.autosale.shop.service;

import com.autosale.shop.model.JwtTokensDTO;
import com.autosale.shop.model.User;

public interface AuthenticationService {

    //SOLID
    // S - Single responsibility
    // сервіс аутентифікації реалізує функціонал тільки аутентифікації

    JwtTokensDTO loginByUserCredentials(User user);

    JwtTokensDTO loginByRefreshToken(String token);

}

File: src/main/java/com/autosale/shop/service/AuthenticationService.kt
package com.autosale.shop.service

import com.autosale.shop.model.JwtTokensDTO
import com.autosale.shop.model.User

interface AuthenticationService {
    fun loginByUserCredentials(user: User): JwtTokensDTO
    fun loginByRefreshToken(token: String): JwtTokensDTO
}
